src/v/cloud_storage/remote_segment.cc

```diff
@@ -879,16 +879,30 @@ class remote_segment_batch_consumer : public storage::batch_consumer {
               "remote_segment_batch_consumer not initialized",
               _parent._seg->get_ntp());
 
-            _parent._cur_ot_state->get().add_gap(
-              header.base_offset, header.last_offset());
-            vlog(
-              _ctxlog.debug,
-              "added offset translation gap [{}-{}], current state: {}",
-              header.base_offset,
-              header.last_offset(),
-              _parent._cur_ot_state);
-
-            _parent._cur_delta += header.last_offset_delta + model::offset{1};
+            if (
+              _parent._cur_ot_state->get().last_gap_offset()
+              < header.base_offset) {
+                _parent._cur_ot_state->get().add_gap(
+                  header.base_offset, header.last_offset());
+                vlog(
+                  _ctxlog.debug,
+                  "added offset translation gap [{}-{}], current state: {}",
+                  header.base_offset,
+                  header.last_offset(),
+                  _parent._cur_ot_state);
+
+                _parent._cur_delta += header.last_offset_delta
+                                      + model::offset{1};
+            } else {
+                // This can happen if we're dealing with overlapping segments
+                vlog(
+                  _ctxlog.debug,
```

Comment:
If we don't expect this to happen aside from with bugs, then this should be an ERROR severity, so that it'll fail ducktape tests if it happens (and be noticeable in production alerting)